craft_archives/repo/apt_sources_manager.py

```diff
@@ -38,6 +39,12 @@
 
 _DEFAULT_SOURCES_DIRECTORY = Path("/etc/apt/sources.list.d")
 _DEFAULT_SIGNED_BY_ROOT = Path("/")
+_COMPATIBLE_ARCHITECTURES = {
+    "amd64": ["amd64", "i386"],
+    "i386": ["amd64", "i386"],
```

Comment:
If your host is amd64 it can support i386 packages too, but I'm not sure the other way around is true. Ignoring hardware that won't support amd64 packages, if I'm running i386 Linux on my otherwise amd64 laptop, I don't think it'll have the right system calls to run amd64 packages?

Reply:
Oop yep, this is certainly a one-directional concept.  Not sure what I was thinking.